cmd/test.go

```
@@ -0,0 +1,205 @@
+		}
+		if imagePath != "" {
+			return fmt.Errorf("Cannot provide both image path and metadata file")
+		}
+	} else {
+		if imagePath == "" {
+			return fmt.Errorf("Please supply path to image or tarball to test against")
+		}
+		if metadata != "" {
+			return fmt.Errorf("Cannot provide both image path and metadata file")
+		}
+	}
+	if len(configFiles) == 0 {
+		return fmt.Errorf("Please provide at least one test config file")
+	}
+	return nil
+}
+
+func RunTests() []*unversioned.FullResult {
+	fullResults := make([]*unversioned.FullResult, 0)
+	for _, file := range configFiles {
+		tests, err := Parse(file)
+		if err != nil {
+			logrus.Fatalf("Error parsing config file: %s", err)
+		}
+		fullResults = append(fullResults, &unversioned.FullResult{
+			FileName: file,
+			Results:  tests.RunAll(),
+		})
+	}
+	return fullResults
+}
+
+func Parse(fp string) (types.StructureTest, error) {
+	testContents, err := ioutil.ReadFile(fp)
+	if err != nil {
+		return nil, err
+	}
+
+	// We first have to unmarshal to determine the schema version, then we unmarshal again
+	// to do the full parse.
+	var unmarshal types.Unmarshaller
+	var strictUnmarshal types.Unmarshaller
+	var versionHolder types.SchemaVersion
+
+	switch {
+	case strings.HasSuffix(fp, ".json"):
+		unmarshal = json.Unmarshal
+		strictUnmarshal = json.Unmarshal
+	case strings.HasSuffix(fp, ".yaml"):
+		unmarshal = yaml.Unmarshal
+		strictUnmarshal = yaml.UnmarshalStrict
+	default:
+		return nil, errors.New("Please provide valid JSON or YAML config file")
+	}
+
+	if err := unmarshal(testContents, &versionHolder); err != nil {
+		return nil, err
+	}
+
+	version := versionHolder.SchemaVersion
+	if version == "" {
+		return nil, errors.New("Please provide JSON schema version")
+	}
+
+	var st types.StructureTest
+	if schemaVersion, ok := types.SchemaVersions[version]; ok {
+		st = schemaVersion()
+	} else {
+		return nil, errors.New("Unsupported schema version: " + version)
+	}
+
+	strictUnmarshal(testContents, st)
```

if strictUnmarshal = json.Unmarshal you're ignoring an error

---

checking it now

---

turns out checking this error actually causes the tests to fail....I didn't write this so I'm not totally sure why, but leaving as is for now